Source/WebCore/page/scrolling/ScrollingStateScrollingNode.cpp

```diff
@@ -286,8 +287,9 @@ void ScrollingStateScrollingNode::setMouseIsOverContentArea(bool flag)
     setPropertyChanged(Property::ContentAreaHoverState);
 }
 
-void ScrollingStateScrollingNode::setMouseMovedInContentArea()
+void ScrollingStateScrollingNode::setMouseMovedInContentArea(const MouseLocationState& mouseLocationState)
 {
+    m_mouseLocationState = mouseLocationState;
```

Comment:
This is one place we could optimize and avoid scrolling tree transactions on every mouse move.

If we only care about mouse moves when over the scrollbar, we should have `AsyncScrollingCoordinator::setMouseMovedInContentArea` always use some placeholder value (-1, -1) when the mouse is not over a scrollbar, and then this function could early return if m_mouseLocationState is unchanged.

Reply:
This is a good idea, but the mouseMovedInContent area event also has the affect of keeping the scrollbar visible while those events are still coming in. I think this would be a good backup if we discover that the extra scrolling tree transactions are a big perf hit.